documentation/Database-Schema-Template.md

```diff
@@ -0,0 +1,75 @@
+# Database Schema Template
+
+![Database Schema](images/database-schema-template.png)
+
+## Tables description: Template
+
+### Version
+When the Admin user wants to edit an existing <b>application template</b> a new version is created. This new version is linked to a duplication of all records of the current <b>application template</b>. The Admin can make changes until publishing this new version.
+With templates versioning we keep previous finalised applications linked to the correct versions of the application templates used. 
+
+* <b>template version</b>
+The `is_current` is set to `'false'` while the version is being created or edited or for all versions that is not the current.  
+The `is_current` is set to `'true'` when the new version of one application template is published. 
+When a new version is added every new application of this application template will be linked to the new version.
```

Comment:
This can be discussed further, as you have pointed out on our weekly planning meeting.
My assumption is that, if we call `template` the `template_version` and the `template_version` as `template`, the entity `template` is only a number and points to a `template_version`.  I think it's add that the template isn't where the template related fields are stored. Thinking of a relational data, it seems strange to me. But I understand your point. So let's talk more about it.

Reply:
Yeah, I noticed that, but thinking about it, it makes sense that the `template` entity is "only a number". Everything else about it can be changed, the ID is the only constant property. I did think that maybe we could move `name` to that table, so that stays constant, but then again, even the name is editable, right? So yeah, if everything about the template can be "versioned", then it makes perfect sense to me that the only thing in the base `template` table that stays static is its id.

Also, based on that, the `is_current` field is actually in the wrong table. I actually think that whoever wrote the `is_current` into the `template_version` table made the assumption that this table would be for storing the copies/versions, which kind of proves my point. I think it's quite dangerous to leave it as it -- anyone who sees the `template_version` table is going to continue to make this assumption.

Also (sorry to go on 😆) it makes much more sense in the tree heirarchy too: A template (which is known only by its ID) has many versions, which, in turn, have many sections and stages, which, in turn, have many elements, etc..   
It makes no sense the other way -- a template version has many templates, which have many sections, etc... 

Either way the `is_current` field is in the wrong table, since it's the "copies" that need to specify the "Current" one, not the base entity.

Reply:
Updated document section about the **Template version** to be re-written after issue #35 is implemented.